submissions/evgenii-del/oop-exercise/index.js

```diff
@@ -0,0 +1,96 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/evgenii-del/a-tiny-JS-world
+   Web app: http://crooked-discovery.surge.sh/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+    constructor(species, name, gender, saying) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+    }
+
+    toString(object) {
+        return [this.species, this.name, this.gender, this.saying].join(';');
+    }
+}
+
+class Man extends Inhabitant {
+    constructor(name, gender, saying = "Hi!", hands = 2, legs = 2) {
+        super('human', name, 'male', saying);
+        this.hands = hands;
+        this.legs = legs;
+    }
+
+    toString(object) {
+        return super.toString() + [this.hands, this.legs].join(';');
+    }
+}
+
+class Woman extends Inhabitant {
+    constructor(name, gender, saying = 'Hello!', hands = 2, legs = 2) {
+        super('human', name, 'female', saying);
+        this.hands = hands;
+        this.legs = legs;
+    }
+
+    toString(object) {
+        return super.toString() + [this.hands, this.legs].join(';');
+    }
+}
+
+class Cat extends Inhabitant {
+    constructor(name, gender, saying = 'meow-meow!', legs = 4) {
+        super('cat', name, gender, saying);
+        this.legs = legs;
+    }
+
+    toString(object) {
+        return super.toString() + this.legs;
+    }
+}
+
+class Dog extends Inhabitant {
+    constructor(name, gender, saying = 'woof-woof!', legs = 4) {
+        super('dog', name, gender, saying);
+        this.legs = legs;
+    }
+
+    toString(object) {
+        return super.toString() + this.legs;
+    }
+}
+
+const inhabitants = [
+    new Man('John'),
+    new Woman('Emma'),
+    new Cat('Sam', 'male'),
+    new Dog('Bob', 'female')
+];
+
+inhabitants.forEach(obj => print(obj));
```

Comment:
Bonus learning for you: try `.forEach(print)`. `print` definition already fits required callback specification

Reply:
It doesn't work for me. I may have misunderstood